tools/rostest/src/rostest/rostest_parent.py

```diff
@@ -48,14 +49,37 @@
 
 class ROSTestLaunchParent(roslaunch.parent.ROSLaunchParent):
 
-    def __init__(self, config, roslaunch_files, port=0):
+    def __init__(self, config, roslaunch_files, port=0, local=False, clear=False):
         if config is None:
             raise Exception("config not initialized")
         # we generate a run_id for each test
-        run_id = roslaunch.core.generate_run_id()
+        if local:
+            param_server = rosgraph.Master('/roslaunch')
+            have_master = False
+            try:
+                run_id = param_server.getParam('/run_id')
+                have_master = True
+            except:
+                run_id = roslaunch.core.generate_run_id()
```

Comment:
roslaunch only busy-waits if the `--wait` command line option in specified (which sets `options_wait_for_master`). rostest does not have a `--wait` option, so I've omitted that behavior here.

otherwise, both rostest and roslaunch will spawn a new, default master to use for the run.

I'll add a generic exception catch here and print a warning if the master could not be reached.